opacus/accountants/gdp.py

```diff
@@ -4,22 +4,23 @@
 
 class GaussianAccountant(IAccountant):
     def __init__(self):
-        self.noise_multiplier = None
-        self.sample_rate = None
-        self.steps = 0
+        self.steps = []
 
     def step(self, *, noise_multiplier: float, sample_rate: float):
-        if self.noise_multiplier is None:
-            self.noise_multiplier = noise_multiplier
-
-        if self.sample_rate is None:
-            self.sample_rate = sample_rate
-
-        if noise_multiplier != self.noise_multiplier or sample_rate != self.sample_rate:
-            raise ValueError(
-                "Noise multiplier and sample rate have to stay constant in GaussianAccountant."
-            )
-        self.steps += 1
+        if len(self.steps) >= 1:
```

Comment:
And correspondingly, in here 😊 (I know that's not part of GDP, but it would be nice if you could make this change here now that you are here!)
https://github.com/pytorch/opacus/blob/eedb4447b4b5271591c268c6f280823b2415e134/opacus/accountants/rdp.py#L14